code/src/components/scroll-to-top-button/ScrollToTopButton.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from "react";
+
+import { ArrowUpward } from "@mui/icons-material";
+
+import AppBox from "@/components/app-box/AppBox";
+import AppIconButton from "@/components/app-icon-button/AppIconButton";
+
+import cn from "@/utils/cn/cn";
+
+import "@/components/scroll-to-top-button/ScrollToTopButton.scss";
+
+const SCROLL_THRESHOLD = 300;
+
+const ScrollToTopButton = () => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  const toggleVisibility = () => {
+    if (window.scrollY >= SCROLL_THRESHOLD) {
+      setIsVisible(true);
+    } else {
+      setIsVisible(false);
+    }
```

Comment:
We can simplify it with `setIsVisible(window.scrollY >= SCROLL_THRESHOLD)`

Reply:
fixed